docs/network-binding-plugin.md

```diff
@@ -357,3 +357,111 @@ The expected artifacts include:
 > **Note**: In order to consume a sidecar, the image needs to be available in
 > an accessible registry.
 > Cluster admins may need to make the sidecar image available on their registry.
+
+## The Complete Plugin
+
+We have covered so far network binding plugins that utilize core domain attachment
+types and/or custom sidecars. While many network bindings can find these
+integration points sufficient, there is a missing piece.
+
+Network bindings that require pod network changes, should use a CNI plugin.
+
+### Using a custom CNI plugin
+
+A CNI plugin is a program that is called by the runtime when the pod is created.
+The program is expected to interpret a standard CNI format input and perform
+changes in the pod network namespace.
+
+> **Note**: The CNI plugin discussed in this section is used for implementing
+> a network binding plugin. It is not to be confused with the CNI plugin that
+> connects the pod to a network on the node.
+
+#### Custom CNI plugin requirements
+
+In order to call multiple times CNI plugins, the cluster requires
```

Comment:
Did you mean - "In order to call multiple CNI plugins"?

Reply:
To call multiple times the same plugin is also included. We can have multiple interfaces with the same binding.